Framework/DataHandling/src/NXcanSASHelper.cpp

```diff
@@ -253,6 +253,17 @@ template <typename T> class QxExtractor {
 } // namespace
 
 namespace Mantid::DataHandling::NXcanSAS {
+
+std::string prepareFilename(std::string &baseFilename, int index, bool isGroup) {
+  if (auto const &extPos = baseFilename.find(NX_CANSAS_EXTENSION); extPos != std::string::npos) {
+    baseFilename.resize(extPos);
+  }
+  auto const addDigit = [&](int index) {
+    return isGroup ? (index < 10 ? "0" + std::to_string(index) : std::to_string(index)) : "";
+  };
+  return baseFilename + addDigit(index) + ".h5";
```

Comment:
It looks like we could use the `NX_CANSAS_EXTENSION` we defined on L33 here. 

It may also be worth editing this method and the parameter to `saveSingleWorkspaceFile` to handle `baseFilename` as a `std::filesystem::path`. This would allow us to use things like `has_extension` to only append the `.h5` extension (with `replace_extension`) if the user hasn't supplied one already.